core/src/main/java/com/hcl/ecomm/core/servlets/ProductDetailsServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.hcl.ecomm.core.servlets;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.hcl.ecomm.core.services.ProductService;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+@Component(service = Servlet.class,
+        property = { "sling.servlet.paths=/bin/hclecomm/productDetails",
+        "sling.servlet.method=" + HttpConstants.METHOD_GET, "sling.servlet.extensions=json" })
+public class ProductDetailsServlet extends SlingSafeMethodsServlet {
+
+    private static final long serialVersionUID = 849553733784661541L;
+    private static final Logger LOG = LoggerFactory.getLogger(ProductDetailsServlet.class);
+
+    @Reference
+    ProductService productService;
+
+    @Override
+    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
+            throws ServletException, IOException {
+        LOG.info("Inside  ProductDetailsServlet doGet Method");
+
+        try {
+         
+		String  sku = request.getParameter("sku");
+		
+        JsonObject productResponse = productService.getProductDetail(sku);
```

Comment:
@Rajamouli-hcl  put a condition to check sku is not null/empty before calling magento service.

Reply:
changes updated as per your comment